### ISSUE TITLE
Disappearing Panel Fonts

### ISSUE THREAD

**Issue by mfreeman72** [linuxmint/Cinnamon#3442]
On a few of my LM systems, I'm seeing a problem with fonts on the Cinnamon panel. On occasion, some of the fonts (but not all) will either disappear (leaving blank space), or turn into solid rectangles/squares. This leaves panel text nearly unreadable in some cases. I've only seen this on systems running the AMD proprietary drivers. It does not happen using the open-source AMD/ATI driver. I had initially thought it was triggered by running WINE, as it appeared most often when running WINE software. However I just saw it happen on a system where WINE isn't installed. So, I'm not sure what triggers it. This is only affecting fonts on the panel. Other fonts on the desktop, windows, etc. are unaffected.

**Comment by mfreeman72** [comment]
I can confirm this is still present in Cinnamon 2.4.

**Comment by celebithil** [comment]
i confirm too ( Cinnamon 2.4.6 )

**Comment by sruggier** [comment]
I just tried Cinnamon 2.2.16-5 in Debian Jessie for the last hour. I can reproduce this problem by activating the scaling effects (in other words, show all windows or show all workspaces). If I don't activate one of those, it doesn't seem to happen. Restarting cinnamon temporarily relieves the symptoms, as it were.

**Comment by xhpohanka** [comment]
Hi. I have similar issue using Cinnamon 2.4.7 on Arch Linux on ThinkPad T410 with Intel video card (i915 driver). I still not found a way to perfectly reproduce it, but the text disappears quite offten. There is also no way (known) to recover without restarting the session. It seems to me that it happens more often when I use external monitor. I also was suspicious about applications like Google Chrome or Eclipse, but probably there is no connection

Please check attached images.

![cinnamon_missing_texts01](https://cloud.githubusercontent.com/assets/2233252/7114665/ea14d326-e1df-11e4-8b1b-fc1187af0d8d.png)
![cinnamon_missing_texts02](https://cloud.githubusercontent.com/assets/2233252/7114664/ea127df6-e1df-11e4-91d8-7b3fa114f578.png)

**Comment by xhpohanka** [comment]
Just an update. It seems that in my case (Intel i915 driver) the issue is reproduced always when going from locked screen on external monitor.

**Comment by xhpohanka** [comment]
I can also confirm @sruggier observation about scaling effects. It is not in 100% but very often. Also some 3rd party applications help to force issue with font disappearing. Krusader is an example in my case.

BTW: Is there any cinnamon developer who could reproduce this problem? I would like to provide more information but there is nothing uncommon in logs.

**Comment by xhpohanka** [comment]
I found another program that seems to help to triggering the issue: **Meld**.

**Comment by Xiol** [comment]
This happened to me after pushing PrintScreen whilst inside a Windows VM running on Virtualbox.

All my windows were minimised and I've lost text on the cinnamon menu, clock, taskbar, etc.

**Comment by Flyweightt** [comment]
Cinnamon 2.2.16 on Debian amd64 here. selecting the fonts menu via the desktop and then change a random setting and back, brings the fonts back.  
This is what i did and the log is here.


file= $home/.xsession-errors


(nm-applet:23425): libnm-glib-CRITICAL **: nm_secret_agent_register: assertion 'priv->registered == FALSE' failed
Window manager warning: Trying to remove non-existent custom keybinding "magnifier-zoom-in".
Window manager warning: Trying to remove non-existent custom keybinding "magnifier-zoom-out".
      JS LOG: Cinnamon started at Tue May 19 2015 15:36:18 GMT+0200 (CEST)

(cinnamon:23413): St-CRITICAL **: st_widget_get_theme_node called on the widget [0x7f7ff828b360 StBoxLayout.menu-selected-app-box] which is not in the stage.
      JS LOG: network applet: Cannot find connection for active (or connection cannot be read)
      JS LOG: network applet: Cannot find connection for active (or connection cannot be read)
Window manager warning: Log level 16: Symbolic icon /usr/share/cinnamon/applets/notifications@cinnamon.org/empty-notif-symbolic.svg is not in an icon theme directory
Window manager warning: Log level 16: Symbolic icon /usr/share/cinnamon/applets/windows-quick-list@cinnamon.org/windows-quick-list-symbolic.svg is not in an icon theme directory
Window manager warning: Log level 16: STACK_OP_ADD: window 0x2400006 already in stack
Window manager warning: Log level 16: STACK_OP_ADD: window 0x2400006 already in stack
      JS LOG: Invalid network device type, is 14
      JS LOG: network applet: Found connection for active
      JS LOG: network applet: Found connection for active
Window manager warning: Log level 16: Symbolic icon /usr/share/cinnamon/applets/windows-quick-list@cinnamon.org/windows-quick-list-symbolic.svg is not in an icon theme directory
Window manager warning: Log level 16: Symbolic icon /usr/share/cinnamon/applets/notifications@cinnamon.org/empty-notif-symbolic.svg is not in an icon theme directory
Window manager warning: Log level 16: Symbolic icon /usr/share/cinnamon/applets/notifications@cinnamon.org/normal-notif-symbolic.svg is not in an icon theme directory
Window manager warning: Log level 16: value "-71.000122" of type 'gfloat' is invalid or out of range for property 'width' of type 'gfloat'
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
Window manager warning: Log level 16: _XEMBED_INFO property has wrong type

Window manager warning: Log level 8: meta_display_register_x_window: assertion 'g_hash_table_lookup (display->window_ids, xwindowp) == NULL' failed
Window manager warning: Invalid WM_TRANSIENT_FOR window 0x32000a8 specified for 0x32000a6 (Initializi).
Window manager warning: Log level 8: meta_display_register_x_window: assertion 'g_hash_table_lookup (display->window_ids, xwindowp) == NULL' failed
Window manager warning: Log level 8: meta_display_register_x_window: assertion 'g_hash_table_lookup (display->window_ids, xwindowp) == NULL' failed
Window manager warning: Log level 8: meta_display_unregister_x_window: assertion 'g_hash_table_lookup (display->window_ids, &xwindow) != NULL' failed
Window manager warning: Log level 8: meta_display_unregister_x_window: assertion 'g_hash_table_lookup (display->window_ids, &xwindow) != NULL' failed
Window manager warning: Log level 16: STACK_OP_RAISE_ABOVE: sibling window 0x32000fc not in stack
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
Window manager warning: Log level 16: Symbolic icon /usr/share/cinnamon/applets/windows-quick-list@cinnamon.org/windows-quick-list-symbolic.svg is not in an icon theme directory
Window manager warning: Log level 16: Symbolic icon /usr/share/cinnamon/applets/notifications@cinnamon.org/normal-notif-symbolic.svg is not in an icon theme directory
Window manager warning: Log level 16: Symbolic icon /usr/share/cinnamon/applets/windows-quick-list@cinnamon.org/windows-quick-list-symbolic.svg is not in an icon theme directory
Window manager warning: Log level 16: Symbolic icon /usr/share/cinnamon/applets/notifications@cinnamon.org/normal-notif-symbolic.svg is not in an icon theme directory
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
Could not find bluetooth module; is the cinnamon-control-center package installed?
Loading Backgrounds module
__init__ took 204.843 ms
Loading Fonts module
Window manager warning: Log level 16: Symbolic icon /usr/share/cinnamon/applets/notifications@cinnamon.org/normal-notif-symbolic.svg is not in an icon theme directory
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added
      JS LOG: Removing an access point that was never added

**Comment by ManIVIctorious** [comment]
I don't think that there will be any backports to cinnamon 2.2 anymore, it would be far more interesting for the developers to know if this is still an issue with latest cinnamon (2.6 will be released in the next days)

**Comment by Flyweightt** [comment]
Good, point, i will upgrade first and see if the bug persists. thanks for the hint

**Comment by xhpohanka** [comment]
For Cinnamon 2.4.8 I can confirm that the fonts are brought back by changing *Font scaling factor* in settings. However I noticed that sometimes not only fonts disappear but also some parts of graphics. 

Please check this picture of app finder (alt + f2)
![cinnamon_missing_graphics](https://cloud.githubusercontent.com/assets/2233252/7742680/5aa6a8d8-ff92-11e4-97c9-2826c73c17e0.png)

The same situation is with application switcher window, I just cannot take a screenshot when alt+tab is pressed.

**Comment by xhpohanka** [comment]
It seems that the issue is *almost* gone in 2.6.3. I just noticed that the font on desktop icons sometimes change its color from white to black.

**Comment by Neowizard** [comment]
Well, for those that can't upgrade to 2.6.x (read, Debian Jessie users) an odd workaround I found is to set and revert font scaling factor.
For some reason, changing the font scaling factor and reverting that change made the issue go away completely. I can't reproduce it anymore, after being able to consistently reproduce it for months now, regardless of any other attempt.
I just ran:
`gsettings set org.cinnamon.desktop.interface text-scaling-factor 1.1`
`gsettings set org.cinnamon.desktop.interface text-scaling-factor 1`
and that was it.
For some reason, after I ran it my background picture reverted to the default Debian Jessie one...no idea why.

Note that prior to Cinnamon 2.2 the commands to change and revert the scaling factor was (note **gnome** instead of **cinnamon**):
`gsettings set org.gnome.desktop.interface text-scaling-factor 1.1`
`gsettings set org.gnome.desktop.interface text-scaling-factor 1`

**Comment by N1kto** [comment]
Have same issue on cinnamon 2.6.13. Are there any plan to fix it?
![fonts-issue-2](https://cloud.githubusercontent.com/assets/5045863/11587048/25e2b4c2-9a77-11e5-80f9-39fbfddabc9a.png)
![fonts-issue](https://cloud.githubusercontent.com/assets/5045863/11587050/267ac79e-9a77-11e5-8d0a-583f9f08f16e.png)

**Comment by Neowizard** [comment]
@N1kto, I noticed that sometimes that workaround doesn't work.
A few other theme-related modifications that help:
Almost always works (but slowest): 
	gsettings set org.cinnamon.desktop.interface scaling-factor 2
	gsettings set org.cinnamon.desktop.interface scaling-factor 1
works part of the time (almost instantly):
	gsettings set org.cinnamon.desktop.interface text-scaling-factor 1.1 (can probably be omitted)
	gsettings set org.cinnamon.desktop.interface text-scaling-factor 1 
	gsettings set org.cinnamon.desktop.interface gtk-theme temp
	gsettings set org.cinnamon.desktop.interface gtk-theme BlackMATE_mod
	gsettings set org.cinnamon.desktop.interface icon-theme temp
	gsettings set org.cinnamon.desktop.interface icon-theme Fog

**Comment by N1kto** [comment]
@Neowizard you are right. I tried using MATE with hope this issue doesn't happen there. But after a few times of suspending it appeared there too. So I start to think that maybe it has something to do with video card/driver. I have an integrated Intel graphics: `Intel Corporation Broadwell-U Integrated Graphics (rev 09)`. The other guy claiming this issue on [Reddit Mint blog](https://www.reddit.com/r/linuxmint/comments/3o0oji/problem_with_fonts_after_suspend_mint_172_cinnamon/) seems to have the same card. What card do you have?
My co-worker running Mint of the same version as I do with Cinnamon as well doesn't have such issue (Nvidia card).

**Comment by Neowizard** [comment]
I'm also using Intel integrated, an "Integrated Intel HD graphics in QM57" (identified by lspci as "Corporation Core Processor Integrated Graphics Controller (rev 02)").

Seems like it could be an issue with the Intel drivers or the way Cinnamon interacts with them.

**Comment by agilob** [comment]
I was experiencing this bug on laptop with Intel GPU. The fonts were disappearing when starting a Qt5 application with a tray icon. A solution I found is to change font anti-aliasing settings, no need to modify shell rc. 3 weeks without the bug.

**Comment by agilob** [comment]
To provide some info, it doesn't look like it's a problem related directly to cinnamon, after having this issue for quite long, I gave up and switched to xfce4, removed all cinnamon dependencies... but the problem still persists. Most people in this thread and in https://github.com/linuxmint/Cinnamon/issues/3732 use Intel GPU. Apparently cinnamon is triggering (more often than Xfce) this bug in other underlying software.

**Comment by Neowizard** [comment]
agilob, could you post some images of the problem under xfce4? I wonder how would it look like with those different panels.

**Comment by agilob** [comment]
@Neowizard I'll make screenshots when this happens next time.

**Comment by LuckyType** [comment]
Hello, i have the same Bug on Debian Jessie 8.5 with Cinnamon 2.2.16. I have 3 computers running Cinnamon, but only 2 of them have this Bug. The other one runs fine.
It happens randomly and up to 10 times a day, only workaround that is working is after the fonts and icons disappear, pressing alt+f2 and typing 'r' or going into tty / terminal and typing cinnamon --replace
Should i try to upgrade my cinnamon? Im a little afraid that it will make the computers even worse. 
The computers with the bug have a Intel Graphics Chip, mine has a nVidia Card

**Comment by N1kto** [comment]
I kind of agree with @agilob's  assumption that this has something to do with graphics rather than with Cinnamon. My colleague is running same Linux mint setup as I do, but on a laptop with dedicated nVidia card and has no issue.
Also recently I discovered another weird behavior resulting in cinnamon crash:
1. Open Chrome's dev tools and go to 'Sources' tab.
2. Now, on the left sidebar you see the sources folders. Try to drag any file in that tree as if you want to move it from one folder to another.
There is action doesn't cause any crash on any laptop i tried except of mine... :)

**Comment by JBYoshi** [comment]
I'm having the same issue. I'm using Cinnamon version 3.0.7. My graphics card is specified as "Intel Corporation 3rd Gen Core processor Graphics Controller", and my CPU is "Intel Core i3-3120M CPU @ 2.50 GHz x 2".

**Comment by shuaimiaomi** [comment]
Confirmed again, Cinnamon 3.07 with Mint 18. AMD Radeon x1600 Mobility, using open source drivers (fglrx unavailable in Mint 18 and AMD has strongly advised against using them). Happens spontaneously, and can't fix it.
![screenshot from 2016-10-30 15-17-52](https://cloud.githubusercontent.com/assets/12941327/19841344/bf01f136-9ee0-11e6-9331-292616a0b00a.png)

**Comment by jedediahfrey** [comment]
Cinnamon on Ubuntu 16.04.

![screenshot from 2016-11-05 11-43-42](https://cloud.githubusercontent.com/assets/223258/20031406/4ad6a6be-a34d-11e6-9855-5777278b642e.png)

**Comment by Redsandro** [comment]
Duplicates:

* https://github.com/linuxmint/Cinnamon/issues/5607
* https://github.com/linuxmint/Cinnamon/issues/5686

---

@Redsandro [said](https://github.com/linuxmint/Cinnamon/issues/5607#issuecomment-256293797):
> @feren 9 out of 10 times I can successfully run `r` to restart Cinnamon as a workaround. E.g.:
> 
> <kbd>Alt</kbd> + <kbd>F2</kbd>, <kbd>r</kbd>, <kbd>Enter</kbd>

Today it didn't work with Geany. Still getting dialogs like this:

![image](https://cloud.githubusercontent.com/assets/1702193/20304417/5e07c156-ab30-11e6-89bf-fe7c461b9130.png)

**Comment by chrisonhub** [comment]
What kernel version are you all running?  On LM i have only seen this issue when using a 4.4 series kernel.  My laptop has intel integrated graphics - interesting to hear that it affects nVidia and AMD/ATI.

My solution for LM17.3 is to stick on the 3.13 kernel series.

**Comment by Neowizard** [comment]
I'm running Cinnamon under Debian Jessie. my Kernel's version is 3.16

**Comment by Redsandro** [comment]
@chrisonhub 
> What kernel version are you all running?

`4.4.0-47-generic` 68-Ubuntu SMP Wed Oct 26 19:39:52 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

**Comment by chrisonhub** [comment]
@Neowizard 
interesting - I didnt see this issue with 3.19 (linux mint 17.1 and 17.2).  When i upgraded to kernel 4.4.0 on LM17.3 and also LM18.0, I regularly saw these graphical problems after waking from sleep or switching between active user sessions.

@Redsandro 
I rolled back to LM17.3 (thank god for backups!) after trying LM18.  I also saw the "invisible mouse" problem, which is one of the known issues with LM18 and intel graphics.

Very disappointed - I was looking forward to the new cinnamon and 16.04 ubuntu base.  I cannot upgrade until this is fixed and it looks like the bugs have been around since 2015 so I have little hope in them being fixed quickly.  Thankfully LM17 is supported for another couple of years.

**Comment by feren** [comment]
This also affects icons 'painted' by Cinnamon in certain circumstances

**Comment by chrisonhub** [comment]
@feren thanks for bringing these threads together.  This bug has been around for 2 years now, any idea who can help fix it?

**Comment by feren** [comment]
Not at all sure

**Comment by Redsandro** [comment]
Linux Mint 18.1 is out. I'll upgrade when I have the time. Perhaps it fixes the infinite shutting down of doom bug. Rolling back is never a good workaround when other bugs are only fixed when moving forward.

**Comment by chrisonhub** [comment]
Someone (Patrick G) suggested that removing the xserver-xorg-video-intel package worked, I got a message but dont think their post is visible on the website.

@Redsandro when you upgrade, please report back on LM18.1 and if removing this package helps.

**Comment by DavidNemeskey** [comment]
I am on Linux Mint 18.1, and I am experiencing the same issue: no text on Cinnamon applets after a suspend.

@chrisonhub I would try that, but could you provide some instructions on what to expect? If I remove the graphics driver, and end up in text mode, then sure, I will be unaffected by the bug, but it still doesn't sound like a solution. :smile:

**Comment by chrisonhub** [comment]
@DavidNemeskey supposedly removing this driver will allow linux to fall-back to the other video driver built into the kernel.  it was mentioned on the Arch forums, I cant remember where at the moment.

What will happen?  Maybe you will be dumped to text mode.  Maybe this issue will be magically fixed.  Someone is going to have to try it and report back!

If you do try this, make sure you have a backup.
If you get dumped to text mode, then surely you could use apt to reinstall the xserver package and reboot?  

LM18 was so bad for me I am still running 17.3 - I have a second hard disk i can swap into my laptop to try this all out on, but i have been too busy!

**Comment by DavidNemeskey** [comment]
@chrisonhub OK, removing the package indeed solved the problem. But what am I using now, then? :smile:
```
$ sudo lshw -c video
[sudo] password for david: 
  *-display               
       description: VGA compatible controller
       product: 3rd Gen Core processor Graphics Controller
       vendor: Intel Corporation
       physical id: 2
       bus info: pci@0000:00:02.0
       version: 09
       width: 64 bits
       clock: 33MHz
       capabilities: msi pm vga_controller bus_master cap_list rom
       configuration: driver=i915 latency=0
       resources: irq:30 memory:f7800000-f7bfffff memory:d0000000-dfffffff ioport:f000(size=64)

$ uname -a
Linux XXX 4.4.0-47-generic #68-Ubuntu SMP Wed Oct 26 19:39:52 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by dqgls** [comment]
_Purging the `xserver-xorg-video-intel` package did NOT solve the bug for me and introduced regressions._ I'm running Linux Mint 18.1 Cinnamon 64bit and have no custom configuration files in `/etc/X11/`.

---

Here's what the package description says:
```
$ apt-cache show xserver-xorg-video-intel
 The use of this driver is discouraged if your hw is new enough (ca.
 2007 and newer). You can try uninstalling this driver and let the
 server use it's builtin modesetting driver instead.
```
My graphics hardware is from 2010.
`sudo lshw -c video` output is no different before and after removing the package. `inxi -G` no longer display any loaded drivers (before: intel), just the unloaded ones which haven't changed (fbdev, vesa).
The regression after purging the package is in the form of a screen flicker, as if caused by bad vsync. I can reproduce the regression consistently in this way: switch to a gnome-terminal window and immediately type something, the whole screen flickers.
There may be other regressions as well, I've been running without the package only for a short time, but will probably install it back soon to get rid of the flicker.

---

Disappearing text on the panel and in applet menus still persists regardless. __I can reproduce the bug__ every time within seconds by switching between songs in Audacious _while MPRIS 2 Server is enabled_ in the Audacious plugin settings (required for player integration with the sound applet), and it doesn't make a difference whether the sound applet actually has or does not have "Control Players" setting enabled. I've tried the same song-switching test in VLC and Xplayer and the bug doesn't happen.
Something must be happening in the interaction between Cinnamon and Audacious' specific use of MPRIS. It seems pretty strange that MPRIS alone would cause a graphical bug in Cinnamon's panel.

>The Media Player Remote Interfacing Specification is a standard D-Bus interface which aims to provide a common programmatic API for controlling media players.

https://specifications.freedesktop.org/mpris-spec/latest/

I completely forgot this bug exists when I was still using 17.3 just two weeks ago because I had disabled MPRIS server in Audacious on that system for some reason, and nothing else triggered it in the whole year since 17.3 was released.

**Comment by Sashz** [comment]
Sorry for my English.
Every time I use Clementine or Rhythmbox this problem occurs: the panel letters disappear. And the interface of some applications also disappears, such as Transmission and System Monitor.
Today the computer crashed with the Clementine application.
I'm leaving Ubuntu for Debian Cinnamon, but this problem is bothering me a lot. I just can not hear music.

**Comment by chrisonhub** [comment]
@dqgls can you check what Kernel you are running in Mint 17.3?  The only one still getting security updates is the 3.13 series and this is running really well for me.  I see this bug when using newer versions of the Kernel.  Try rolling back to the 3.13.102 or whatever is the newest and see if the bug persists.

**Comment by dqgls** [comment]
@chrisonhub I experienced the issue straightway when LM 17.3 came out in December 2015, so it's not just newer kernels, although I think the first kernel on Rosa was 3.16 so I will try 3.13 later.

If I remember correctly it was LM 17.3 that introduced "Control players" feature with Cinnamon 2.8. At that time I immediately noticed a correlation between song change in Audacious and DE integration (desktop notifications, album art change in sound applet, etc...), so I just turned off a bunch of stuff instead of investigating more closely, and then completely forgot about the issue which never appeared ever again.

Switching from 17.3 to 18.1 I decided to enable player integration in the sound applet, and everything was fine for a while, but then it started appearing again fairly often.

Some albums trigger it after a few song changes, others almost never, and after some testing I'm now fairly certain that it's connected to album art data (song tags, whether normal or unusual, don't make any difference whatsoever). But it's not just a few albums either. Note that album art can be embedded into audio file metadata itself instead of existing as a separate image file. So far albums that cause problems are such.

I tried putting `COGL_ATLAS_DEFAULT_BLIT_MODE=framebuffer` in `/etc/environment` (see https://bugzilla.novell.com/show_bug.cgi?id=913425#c7), and then couldn't reproduce the issue at all with those same albums that trigger it quickly. But after about 24 hours of uptime the issue reappeared and could again be reproduced within seconds on each attempt.

Right now I'm back on 17.3, which has the same pattern as 18.1: the issue appears only if MPRIS is enabled, anything else doesn't make a difference. Desktop notifications don't cause it either. What I find strange is that, as far as I understand, MPRIS itself has nothing to do with graphics manipulation (it only sends album art as any other data), and that it's not just Audacious but Clementine too, and @Sashz reports Rhythmbox as well.

And to repeat again, whether "control players" feature is enabled or not, whether album art is actually shown on the panel or not, all this makes no difference. The reason why I suspect sound applet regardless is that when a player has MPRIS enabled Cinnamon doesn't let that player's process die until Cinnamon itself is restarted via `--replace` option.

When I have more time I will test Audacious + MPRIS on 17.2 as well, which I don't think had the sound applet integration yet.

@Sashz can you do the following:
- look around Rhythmbox for any settings that mention MPRIS, remote control, servers, etc., and try turning off those settings
- if you suspect certain albums or songs then look if they have embedded album art: navigate to their directory and see if those audio files themselves have album art thumbnails

**Comment by chrisonhub** [comment]
@dqgls I just grabbed an album from Amazon MP3, which appears to have album art embedded.  I only have Banshee installed, but it does have the MPRIS-dbus plugin active.  They didnt trigger the bug for me, but I only did limited testing.   I am on Kernel 3.13.0-100 (latest is -106)

I skipped tracks in Banshee and also using the Cinnamon applet.

**Comment by chrisonhub** [comment]
@dqgls  regarding the flicker, have you tried adding
CLUTTER_PAINT=disable-clipped-redraws:disable-culling
to your etc/environment ?

**Comment by chrisonhub** [comment]
I have stumbled across more information on this bug - people have seen the same issue in MATE, intermittently - [https://bbs.archlinux.org/viewtopic.php?id=186783](url)

One comment in the thread:  found that switching to the UXA backend fixed the problem: [https://wiki.archlinux.org/index.php/Intel_graphics#SNA_issues](url)

In other news, Debian now defaults to using the Modesetting driver, not the intel one:
[https://tjaalton.wordpress.com/2016/07/23/intel-graphics-gen4-and-newer-now-defaults-to-modesetting-driver-on-x/](url)

Mint 17.3 with the 3.13-106 kernel and everything runs fine, no sign of this bug.

**Comment by petermoo** [comment]
LM 18.1 clean install on Intel machine with Intel graphics. I have the problem after switching external screen. I closed the lid on the notebook, unplugged the TV, walked to a different room, plugged in a HDMI monitor, and opened the lid. Problem is only in the Cinnamon bar at the bottom and the menu. Not in any apps.

**Comment by ip413** [comment]
the same problem here, but not every time and on Mate not Cinnamon; once on four, five times.
Config:
Thinkpad T-430s
Release Linux Mint 18 Sarah 64-bit
Kernel Linux 4.4.0-21-generic x86_64
MATE 1.14.1
3rd Gen Core processor Graphics Controller version 0.9 Driver: "i915"

**Comment by mikkorantalainen** [comment]
Can you test with Linux kernel 4.8 or later? Some people are claiming that this has been fixed in kernel version 4.8. You can try installing package linux-image-lowlatency-hwe-16.0 or linux-generic-hwe-16.04.

**Comment by petermoo** [comment]
The problem occurred on my work and study machine. I cannot afford to do anything drastic for a month or two. My current approach is to shutdown before shutting the lid.

**Comment by ip413** [comment]
@mikkorantalainen
Kernel 4.8 seems to work for me. Even internet connection works fine after wake up. For me is almost closed issue (I would like to wait for a week or two to conclude it with certainty).

**Comment by Redsandro** [comment]
(how) Can I test this relatively pain-free or will all kinds of software break? Now running 4.4.0-59-generic.

**Comment by ip413** [comment]
@Redsandro
I've tested by installing linux-image-lowlatency-16.04 - and in general whole distro works fine, and only VirtualBox had some problems. But due the fact that I work with VB in daily routine I removed this version and working right now with old one. And I had version 4.4.0-21-generic - so it was older than yours.
For me it was almost "pain free" :)

**Comment by mainmachine** [comment]
@ip413 - why did you use the lowlatency kernel...?

You should use the standard "generic" kernel, since that's the analogue to the 4.4.0-21 kernel you have been using:

```sudo apt install linux-image-generic-hwe-16.04```

As of right now that will install the 4.8.0-39-generic kernel...

**Comment by mainmachine** [comment]
@mikkorantalainen - Is there a reason to use the lowlatency kernel in this case? Isn't that just throwing in another variable?

**Comment by ip413** [comment]
@mainmachine
I installed this packed without thinking, for me it is "the same thing". But this could be significant, and I tested generic version and for the issue the result is the same - it seems to work fine. But, my wifi connection is dead - lowlatency kernel worked better here.

**Comment by ageorgop** [comment]
4.8.0-41-generic hard locks my laptop.

**Comment by mikkorantalainen** [comment]
Simply running command
    sudo apt install linux-generic-hwe-16.04
in terminal window and restarting the computer should fix the problem. In case of problems, you can select older kernel (4.4.x) from grub boot menu (if system does not work at all) and remove the latest kernel package with
    sudo apt remove linux-generic-hwe-16.04

**Comment by mikkorantalainen** [comment]
The generic 4.8 kernel should be good enough if you just want to avoid the rendering problems. I find that lowlatency kernel works all around better for me (better performance overall in addition to improved latency).

(The last time I had any issues caused by lowlatency kernel was a couple of years ago when I tried to use lowlatency kernel with Virtualbox. Virtualbox kernel drivers have since been fixed and 5.x series works just fine with lowlatency kernel.)

**Comment by mikkorantalainen** [comment]
@ip413 which version of Virtualbox are you using? I'm currently using Virtualbox (version 5.1.16-113841~Ubuntu~xenial) with hwe lowlatency kernel and it works just fine.

**Comment by mikkorantalainen** [comment]
According to https://bbs.archlinux.org/viewtopic.php?pid=1633207#p1633207 it seems that this bug is caused by some unidentified change in Linux kernel 4.3 or 4.4 and fixed around version 4.6. If somebody can easily reproduce this bug and knows how to do kernel bisect (or is willing to learn) pinpointing the exact change might allow fixing the issue in default generic kernel, too. I can reproduce this bug randomly around once in 5 days so I cannot figure the exact cause.

**Comment by ip413** [comment]
@mikkorantalainen 
My VB is 5.1.2 r108956 (Mint/Mate)- the problem is probably only about rebuilding virtualbox-dkms. VB works in fact, but all images are not working at all, I get some awful errors.

**Comment by petermoo** [comment]
4.8.0.41 was sitting there in Linux Mint Update Manager. Click. It is installed. No visible changes or problems. I will test with the plugging in and out of monitors/TVs, etc.

**Comment by ip413** [comment]
I don't have 4.8.0.41 in update manager :) - anyway I tried 4.8 but I can't make my VirtualBox images working. I going back to old, "good" 4.4.0

**Comment by Redsandro** [comment]
Trying out `4.8.0-41-generic #44~16.04.1-Ubuntu SMP` now. Give it a week or so before I can make a judgment.

---

Remember, for VirtualBox support, you need to compile the virtualbox-dkms using linux header files. E.g. (reboot with the `hwe` kernel first and then):
```
sudo apt install linux-headers-generic-hwe-16.04
sudo dpkg-reconfigure virtualbox-dkms
```

~~And reboot your VM. Saved states don't work.~~ Everything works for me, even saved states.

---

@mikkorantalainen is there a performance benefit to VirtualBox using the lowlatency kernel? Or are you in audio?

Can someone, for my sanity, explain what `hwe` means? _"Hlatest Wlinux Ekernel"_?

**Comment by feren** [comment]
@Redsandro **H**ard**w**are **E**nablement Kernel

**Comment by mikkorantalainen** [comment]
HWE = https://wiki.ubuntu.com/Kernel/LTSEnablementStack

The reason lowlatency improves overall system performance is that kernel has more changes to run tasks in paraller. For example, lowlatency kernel can run waiting tasks during short idle periods while waiting for IO device. Generic kernel could also start running the same task but it cannot interrupt task processing as fast as lowlatency kernel and as a result, UI device may end up waiting for CPU time is another task has been started on CPU. If your system runs less paraller tasks than you have CPU cores, this does not apply to you. However, if your system is busy but all tasks are not CPU limited, lowlatency kernel usually improves things.

Technically lowlatency kernel sacrifices some CPU throughput to improve overall system latency. If you're mining bitcoin on CPU or running HPC code generic kernel is better. For the normal people, lowlatency kernel is usually better.

After saying that, generic kernel has probably been tested more so some more exotic hardware may work better with generic kernel. The ability to interrupt pretty much any task any time will put more requirements on driver quality. As a result, poor quality hardware drivers may work better with generic kernel. If kernel driver does not contain bugs, it will work with lowlatency kernel just fine.

**Comment by Flyweightt** [comment]
I am on debian 8 and I added a 4.9 kernel via Jessie backports and as well cinnamon 3. Things look good will wait a week to see if the issue is gone.  apt-get install -t jessie-backports linux-image-.... to get it installed. Of course you have to add Jessie backports to your apt sources.list and run apt-update first. BTW  the rt kernel did not work well at all for me.

**Comment by Redsandro** [comment]
@feren @mikkorantalainen thanks for helping me stay sane. 10 years linux and I had never heard of it. Please don't tell anyone.

> For the normal people, lowlatency kernel is usually better.

This is interesting, because everyone gets generic for free, and only the technically anointed get to do lowlatency.

**Comment by danbanker** [comment]
I have this issue as well.  I see that you have been working on it for years now.  Is there a resolution in site?

**Comment by mikkorantalainen** [comment]
@Flyweightt Did you test -lowlatency or -rt kernel? I'm asking because Ubuntu no longer has -rt kernel at all and -lowlatency kernel is just PREEMPT instead of hard realtime (like the old -rt kernel in older ubuntu versions was).

**Comment by Redsandro** [comment]
The problems seem to be permanently gone using kernel `4.8.0`. However, `ecryptfs` is now broken.

See: https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1682410

**Comment by feren** [comment]
Shall we close this issue once Linux Mint ever gets Kernel 4.8.0? I know that'll be a while, but still...

**Comment by chrisonhub** [comment]
Ubuntu 16.04.2 will be released with Kernel 4.8 and that will be the LTS supported kernel.  I guess that will mean it will become available to LM sometime soon, maybe when 18.2 comes out.

At least for Ubuntu, you can see their Kernel and LTS plans here:
[https://wiki.ubuntu.com/Kernel/RollingLTSEnablementStack](https://wiki.ubuntu.com/Kernel/RollingLTSEnablementStack)

**Comment by raph82** [comment]
I 'm sad to report that I have this issue with a 4.8 kernel :(

From a fresh install of Linux Mint 18.2 Sonya, on a recently acquired Toshiba Satellite A100 laptop (old laptop but new to me, so no older comparison points).

`uname -sr`
`Linux 4.8.0-53-generic`

`cinnamon --version`
`Cinnamon 3.4.3`

`glxinfo | grep -i vendor`
`server glx vendor string: SGI
client glx vendor string: Mesa Project and SGI
    Vendor: X.Org R300 Project (0x1002)
OpenGL vendor string: X.Org R300 Project`

`lspci | grep VGA`
`01:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] RV515/M54 [Mobility Radeon X1400]`

**Comment by petermoo** [comment]
I did an upgrade from my working 18.1/4.8 to 18.2 and the problem did not resurface. Have not tried fresh install. Will not try fresh install until this problem fixed.

**Comment by leigh123linux** [comment]
In two years no one has posted logging or a backtrace for the issue, closing it as 'Insufficient info' is provided!

**Comment by DavidNemeskey** [comment]
@leigh123linux I haven't seen any requests here about logging or backtrace, nor instructions how to produce one. People uploaded e.g. screenshots or the relevant part of the output of `lshw`, so maybe if you had asked for it, people would have provided said info...

**Comment by feren** [comment]
I guess this is fixed in LM 18.2, as I've never got the bug since 18.1...

**Comment by leigh123linux** [comment]
@DavidNemeskey 

A few sceenshots and some random hardware info is useless when trying to find code issues.
We shouldn't be expected to ask for logs to be provided, could you imagine how long it would take to respond to 30+ reports asking for this basic info?
I used to see this issue occasionally with my T410 using Fedora 21 - 22 , the user logs didn't show anything related to the issue.
I haven't seen the issue on the T410 since updating to newer Fedora which indicates an issue in the Ubuntu graphics stack.
FTR I spent time trying the steps provided to reproduce, none of them produced the issue.


For those on mint 18.2 using newer Intel hardware could try removing the intel legacy driver and switch to modesetting driver instead

```
sudo apt-get remove xserver-xorg-video-intel
```

reboot

P.S 18.2 radeon users could also try modesetting driver as well, I have no idea how to switch to it.

https://www.phoronix.com/scan.php?page=news_item&px=Radeon-Modesetting-DDX-Xenial

**Comment by DavidNemeskey** [comment]
@leigh123linux You could always provide a link to a page somewhere (e.g. in the central README.md) that describes what information you would like to see for each report. I would gladly provide any info I can, but it's difficult without knowing what you might be needing.

The logs not providing enough data is of course, another matter altogether. It is also a bummer that you could not reproduce it, but before removing the driver, I could, consistently, with logging in and out, so at that point I could have acquired any info. If it turned out to be an Ubuntu bug, we would have been in a better position to file it upstream.

**Comment by allenerenee** [comment]
I have this issue frequently running LM Cinnamon 18.0 Sarah - usually once or twice a week, almost always when the computer wakes up after opening the lid. Kernel version 4.4, on a Toshiba Satellite C55-B5355. If someone could point me to the location of the log file I will post, but I haven't got the first clue where to find it.

**Comment by raph82** [comment]
As @allenerenee, I'd gladly provide more useful info.
Until instructed how to do so, I'm using the workaround [cited by leigh123linux](https://github.com/linuxmint/Cinnamon/issues/3442#issuecomment-315524259):

`sudo apt-get remove xserver-xorg-video-intel`

**Comment by Redsandro** [comment]
@allenerenee this is fixed in Kernel 4.8 (`linux-generic-hwe`) and 4.10 (`linux-generic-hwe-edge`) as first referenced by @mikkorantalainen 

```
sudo apt install linux-image-generic-hwe-16.04
```

**Comment by petermoo** [comment]
@allenerenee, there is syslog in /var/log/syslog. You can open Nemo, the file lister, and browse syslog using the text editor. Find the time where you closed the list and the time when you opened the lid.

To keep the log small, you could close every application, wait a minute, close the list, open, then select syslog. That section of the log will be free from distracting application related messages.

**Comment by raph82** [comment]
@Redsandro this is not fixed in kernel 4.8, see [one of my previous comment](https://github.com/linuxmint/Cinnamon/issues/3442#issuecomment-315437010).

The workaround [cited by leigh123linux](https://github.com/linuxmint/Cinnamon/issues/3442#issuecomment-315524259) does not work. I tried it for a few days before commenting back and it was fine. Then I used Wine and I suspect it triggered the bug again. Will try to test and post relevant syslog.

**Comment by Redsandro** [comment]
@raph82 I hear you. I wonder if there are two or more different causes for the same problem.

For me the problem does very clearly not occur on 4.8 or 4.10. When I switch back to 4.4, the problem is back immediately. "Something" is fixed for my hardware configuration quite observably so.

**Comment by raph82** [comment]
I've upgraded to a 4.10 kernel a few days ago and since I did not see this bug.

**Comment by raph82** [comment]
Funny how it's only after I post "it works again" messages that bugs resurface.
I've seen it yesterday in Nemo and today on the panel, both times after waking up the computer (reopening the lid).
In syslog I found lines about corrupted memory. The time frames seem to match the bug.
I'm attaching screen captures (including one with others, possibly related, bugs) and logs.
Does this report help?

[toshiba-syslog-2017-08-13.log.txt](https://github.com/linuxmint/Cinnamon/files/1222114/toshiba-syslog-2017-08-13.log.txt)
[toshiba-syslog-2017-08-14.log.txt](https://github.com/linuxmint/Cinnamon/files/1222113/toshiba-syslog-2017-08-14.log.txt)
![capture du 2017-08-13 11-33-21](https://user-images.githubusercontent.com/14821707/29271113-355646d8-80fa-11e7-9618-e9efd71e4e49.png)
![capture du 2017-08-14 13-56-42](https://user-images.githubusercontent.com/14821707/29270809-be0766ee-80f8-11e7-8791-78570c89e597.png)

**Comment by mscheper** [comment]
@raph82: I'm having a pretty severe case of this issue right now, but I don't see the words 'memory' or 'corrupt' in `/var/log/syslog`. Can you tell me the exact messages you're seeing? I'm just curious if your memory issues are related to the problem we're both apparently having.

I just typed `grep -i '\(memory\|corrupt\)' /var/log/syslog /var/log/kern.log /var/log/auth.log /var/log/Xorg.*`, which should cover all the files in `/var/log` that were changed today, and it didn't come up with anything interesting. I don't have a Toshiba, though, so there's no 'toshiba-syslog-*' file on my system.

**Comment by mscheper** [comment]
In case any expert on how Mint or Cinnamon render fonts is watching this bug but not #3811, I refer them to the comment I just posted there. I found that the issue isn't quite as simple as 'sometimes, the number 2 and letter Q stops rendering for a particular font'; the size, colour and whether the window has focus apparently affects which characters are rendered. This is a pretty debilitating issue on my system, so I'm happy to help try to solve it. (It'd be nice to minimise how often I have to reboot my system, though. ☺)

**Comment by allenerenee** [comment]
By the time I got the response how to find it, I'd upgraded to the 4.10 kernel and haven't had this issue since.

On October 12, 2017 10:04:25 AM EDT, Michael Scheper <notifications@github.com> wrote:
>@raph82: I'm having a pretty severe case of this issue right now, but I
>don't see the words 'memory' or 'corrupt' in `/var/log/syslog`. Can you
>tell me the exact messages you're seeing? I'm just curious if your
>memory issues are related to the problem we're both apparently having.
>
>I just typed `grep -i '\(memory\|corrupt\)' /var/log/syslog
>/var/log/kern.log /var/log/auth.log /var/log/Xorg.*`, which should
>cover all the files in `/var/log` that were changed today, and it
>didn't come up with anything interesting. I don't have a Toshiba,
>though, so there's no 'toshiba-syslog-*' file on my system.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/linuxmint/Cinnamon/issues/3442#issuecomment-336147375

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by mscheper** [comment]
Thanks, @allenerenee. I've seen reports that updating to 4.8 doesn't fix the problem, but I'll try 4.10.

**Comment by raph82** [comment]
@mscheper: the (possibly) relevant messages in the [previously](https://github.com/linuxmint/Cinnamon/issues/3442#issuecomment-322174150) attached syslogs are:
`Corrupted low memory`
`Memory corruption detected in low memory`

IIRC those come from /var/log/syslog; toshiba-syslog* was the filenames I gave the extracts.

By the way I'm using 4.10 [since](https://github.com/linuxmint/Cinnamon/issues/3442#issuecomment-322028757) [August](https://github.com/linuxmint/Cinnamon/issues/3442#issuecomment-322174150).

**Comment by Heycharger** [comment]
I just signed up to Github to make posts here...
I am having this trouble of text disappearing on the taskbar and menus - specifically when I use Clementine, the music player.

OS:
Linux Mint 18.2  64
Cinnamon 3.4.6
Kernel 4.10.0-38-generic

Hardware
Toshiba Tecra A11-12F
i3 M330 2.13ghz
Graphics - generic driver - (presumably intel graphics)

One fix, as suggested much earlier in this thread, is to do
       gsettings set org.cinnamon.desktop.interface text-scaling-factor 1.1
followed by
       gsettings set org.cinnamon.desktop.interface text-scaling-factor 1

For some people, it seems that subsequent kernel and other updates solved this, but it's still happening on my system.

**Comment by GuppyBubble** [comment]
Greets! 
As like Heycharger, I joined to post regarding the same issue. My taskbar and menus do exactly the same (screenshot would be same as 'xhpohanka' posted back on  on 13 Apr 2015).

I've done a few time consuming experiments to see where the issue may lie. I'm no tech wizard, although am mildly competent at getting around with the help of Google and various threads, and it seems in my case this issue occurs when I am running any type of audio player. I've tried VLC, Banshee, Clementine and Audacious and ran out of patience.

It's been suggested that this could be a graphics issue. Personally I don't see how this is the root cause as nothing else seems to create the issue. When using VLC to play video files, I've left the application running for a few hours with no change to the taskbar & menus, yet when I play audio (through VLC) the problem arises within 30 minutes of using the application. Again, this issue happens solely with the use of applications when playing audio files (WAV, FLAC & Mp3 specifically).

Not knowing anything about programming or writing scripts/code, I can't suggest a work around. However, may I be so incredibly bold as to suggest that the issue may be a conflict between audio and graphics somewhere (??) This issue never occurred with Ubuntu, only with Mint. Would be nice to get this one solved...


Lenovo ThinkPad-T410
Kernel:      4.10.0-40-generic x86_64 (64 bit gcc: 5.4.0)
Desktop:   Cinnamon 3.4.6 (Gtk 3.18.9-1ubuntu3.3) dm: lightdm Distro: Linux Mint 18.2 Sonya
CPU:         Dual core Intel Core i5 M 520 (-HT-MCP-) cache: 3072 KB
Graphics:  Card: Intel Core Processor Integrated Graphics Controller bus-ID: 00:02.0 chip-ID: 8086:0046
Audio:     Card Intel 5 Series/3400 Series High Definition Audio
           driver: snd_hda_intel bus-ID: 00:1b.0 chip-ID: 8086:3b56
Sound: Advanced Linux Sound Architecture v: k4.10.0-40-generic

**Comment by Shankaraya** [comment]
Clementine produces the scrambled taskbar effect. Years this is bothering me. Different versions of mint and clementine.
Linux mint 17.3 cinnamon, clementine 1.3.1 Thinkpad T410.

**Comment by Shankaraya** [comment]
Clementine produces the scrambled taskbar effect. Years this is bothering me. Different versions of mint and clementine.
Linux mint 17.3 cinnamon 2.8.8, clementine 1.3.1 Thinkpad T410.
LMDE 3 "Cindy" cinnamon 3.8.9 same bug. Some files in clementine and strawberry mess up my menu and taskbar. For listening music cinnamon is a pain. Restarting cinnamon every theme. I like strawberry (clementine fork) and clementine for their database handling. In VLC does not happen, all ok. I don't like mate or Xcfe. It's probably cinnamon bug.
sudo apt-get remove xserver-xorg-video-intel does not change anything.
log says: Error while parsing CSS Style (Line:1 1, character: 38): Using Pango syntax for the font: style property is deprecated; please use CSS syntax
Error while parsing CSS Style (Line:1 1, character: 26): not a number
Don't know if that's connected.
Where to communicate with cinnamon developers?

**Comment by germanfr** [comment]
> log says: Error while parsing CSS Style (Line:1 1, character: 38): Using Pango syntax for the font: style property is deprecated; please use CSS syntax
Error while parsing CSS Style (Line:1 1, character: 26): not a number

@Shankaraya that looks like a theme parsing error caused by a property like `font: bold` or something like that. Can you reproduce with the default theme?

**Comment by Shankaraya** [comment]
Changed to mate desktop. Even deleted cinnamon.

**Comment by PlatinumLucario** [comment]
I can confirm that this issue is not just limited to Cinnamon DE, nor is it limited to just Linux Mint. I can confirm that this bug also occurs on GNOME Shell 3.36.9 and on Ubuntu 20.04.

And to make matters worse, when this bug occurs, it also breaks GNOME Settings function too! Which means if you try to open GNOME Settings, the application will immediately crash, every time the Settings app is opened.

OS Information:
Linux Kernel version: 5.4.0-91-generic
Distribution release: Ubuntu 20.04
Distribution version: 20.04.3 LTS

DE Information:
GNOME version: 3
GNOME Shell version: 3.36.9

Display Server Information:
Display Server: X11
Display Server Version: X.Org 1.20.11

Display Manager Information:


Here's one of the best ways to reproduce this issue:
1. Open Discord
2. Enter into any voice channel on Discord and just stay there for a day, it may range from 30 minutes to 24 hours for it to occur.

Expected results:
The text shouldn't vanish from the desktop environment and it shouldn't cause any graphical issues. Instead, if anything stops working, it should reload!

Actual results:
The text vanishes from the desktop environment and/or other graphical errors occur, and the GNOME settings app crashes upon attempting to open.

I'm also going to copy and paste this into Ubuntu's bug reporter too.
![Screenshot from 2021-12-20 23-53-28](https://user-images.githubusercontent.com/1906189/146784524-df557a28-cf10-4b17-b71a-903ba0eb8416.png)
![Screenshot from 2021-12-20 23-56-57](https://user-images.githubusercontent.com/1906189/146784538-bb48bf2f-218a-406b-a3f0-58b913b6491e.png)
![Screenshot from 2021-12-20 23-57-09](https://user-images.githubusercontent.com/1906189/146784547-1cc8d322-672e-474d-b9a4-14ce6b85a1e5.png)

**Comment by mikkorantalainen** [comment]
> Video Card Information: Hardware: NVIDIA Corporation: GM206 [GeForce GTX 950] Driver: nvidia-driver-470 (proprietary, tested)

As this bug is highly probably about GPU driver issue, if you cannot reproduce the problem with intel integrated graphics, your bug is probably some other issue.

Also, any issue with a test case where the steps include "wait for random delay between 30 min and 24 h" will be hard to fix. Do you have any evidence that it will happen for sure in 24 h?

In addition, the last screenshot shows crash with partial stacktrace suggesting that this crash happens within libglib which is not the software running the GPU nor display so it's probably not related to rendering issue. Have you been able to reproduce this issue with another similar computer to make sure the issue is not e.g. RAM corruption because of hardware issue?
